beta-src/src/components/ui/WDFullModal.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { useState } from "react";
+import { Box, Stack, Button } from "@mui/material";
+import WDInfoDisplay from "./WDInfoDisplay";
+import GameMode from "../../enums/GameMode";
+import Season from "../../enums/Season";
+import Ranking from "../../enums/Ranking";
+import IntegerRange from "../../types/IntegerRange";
+import GameType from "../../enums/GameType";
+import { CountryTableData } from "../../interfaces";
+import Device from "../../enums/Device";
+import WDInfoPanel from "./WDInfoPanel";
+import WDPress from "./WDPress";
+import ModalViews from "../../enums/ModalViews";
+
+interface WDFullModalProps {
+  children: React.ReactNode;
+  countries: CountryTableData[];
+  device: Device;
+  gameMode: GameMode;
+  gameType: GameType;
+  maxDelays: IntegerRange<0, 5>;
+  potNumber: IntegerRange<35, 665>;
+  rank: Ranking;
+  season: Season;
+  title: string;
+  userCountry: CountryTableData;
+  year: number;
+}
+
+const textButtonStyle = {
+  borderRadius: 0,
+  fontWeight: 400,
+  minWidth: 0,
+  p: "0 0 5px 0",
+  "&:hover": {
+    background: "transparent",
+  },
+};
+
+const textButtonSelected = {
+  borderRadius: 0,
+  borderBottom: "solid black 2px",
+  fontWeight: 600,
+  minWidth: 0,
+  p: "0 0 5px 0",
+  "&:hover": {
+    background: "transparent",
+  },
+};
+
+const WDFullModal: React.FC<WDFullModalProps> = function ({
+  children,
+  countries,
+  device,
```

Comment:
i don't get why we are passing device

Reply:
same here. the device props will be passed to the fullModal to affect padding in different device layout; to title section to affect the maxwidth of the lineclamp title when the viewport is wider or narrower. please see the videos below: 



https://user-images.githubusercontent.com/99943146/160154709-7ad13202-e253-4756-b4ef-def86d09ae67.mov

Reply:
maybe we need a hook like useDevice for this instead of passing device?